src/TimeModeling/seismic_preconditioners.jl

```diff
@@ -191,11 +191,21 @@ end
 
 function depth_scaling(m, model)
 # Linear depth scaling function for seismic images
-    m = reshape(m,model.n)
+    m = deepcopy(reshape(m,model.n))
     filter = sqrt.(0f0:model.d[2]:model.d[2]*(model.n[2]-1))
     F = diagm(0=>filter)
-    for j=1:model.n[1]
-        m[j,:] = F*m[j,:]
+    if length(model.n) == 2
```

Comment:
There is probably a nicer way than this if else with joKron.

Reply:
I've made a clean way to do depth scaling no matter 2D or 3D in the latest commit